### PR TITLE
feat: Implemented `TypeEnumRanker`

### PR DESCRIPTION
**Pull request by LukasRonn**

### CODE REVIEW

**Comment by LukasRonn** [comment]
Experiencing some odd behavior. Testing.

**Comment by LukasRonn** [comment]
Works. https://github.com/flix/flix/pull/5753 fixes the odd-behavior.

**Comment by LukasRonn** [comment]
Actually experiencing same issues as in https://github.com/flix/flix/pull/5753 but just with `TypeEnumCompletion` being generated without matching word.

**Comment by LukasRonn** [comment]
https://github.com/flix/flix/pull/5776 fixes problem.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/ranker/TypeEnumRanker.scala [hunk 1]
```diff
@@ -18,16 +18,44 @@ package ca.uwaterloo.flix.api.lsp.provider.completion.ranker
...
+    }
+  }
+
+  /**
+    * Checks if set[SourceLocation] does not contain any SourceKind.Real
+    *
+    * @param set the set of SourceLocations.
+    * @return    true, if none of the SourceKinds are Real, false otherwise.
+    */
+  private def hasNoRealSourceKinds(set: Set[SourceLocation]): Boolean =
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/ranker/TypeEnumRanker.scala:59]: Its generally bad to have functions that are negated. Why not have `hasRealSourceKind` and then use `!` in the call to find?

**Pull request merged**